src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -343,7 +344,11 @@ private void retrievePullRequests(SCMSourceCriteria criteria, SCMHeadObserver ob
                 checkInterrupt();
                 PullRequestSCMHead head = new PullRequestSCMHead(pull.getSource().getRepository().getOwnerName(),
                         pull.getSource().getRepository().getRepositoryName(), repositoryType,
-                        pull.getSource().getBranch().getName(), pull);
+                        pull.getSource().getBranch().getName(), pull,
+                        getRepoOwner().equalsIgnoreCase(pull.getSource().getRepository().getOwnerName())
```

Comment:
Actually you can do this too in GitHub, you just have to go into Settings and rename the fork repository... I suspect that could cause issues that we are not expecting in GitHub Branch Source too